### PR TITLE
Add attributes to build image

### PR DESCRIPTION
**Pull request by alexyueer**
Changes to support attributes below:
- Customer can create instance with image family.
- Customer can set SecurityEnhancementStrategy parameter when create instance.

Reference: https://www.alibabacloud.com/help/doc-detail/25499.htm

### CODE REVIEW

**Comment by alexyueer** [comment]
@SwampDragons @nywilken Could anyone help this?

File: builder/ecs/run_config.go [hunk 1]
```diff
@@ -132,14 +139,22 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
...
 	// Validation
 	errs := c.Comm.Prepare(ctx)
-	if c.AlicloudSourceImage == "" {
+	if c.AlicloudSourceImage == "" && c.AlicloudImageFamily == "" {
 		errs = append(errs, errors.New("A source_image must be specified"))
 	}
 
-	if strings.TrimSpace(c.AlicloudSourceImage) != c.AlicloudSourceImage {
+	if c.AlicloudSourceImage != "" && c.AlicloudImageFamily != "" {
+		errs = append(errs, errors.New("The image_family and source_image only can be set one"))
```
Comment (nywilken) [inline builder/ecs/run_config.go:147]:
```suggestion
		errs = append(errs, errors.New("An image_family and source_image can not be specified at the same time. Please use only one."))
```

**Review by nywilken — APPROVED** [review]
@alexyueer apologies for the delay. We've been focusing on some changes to Packer core so we are reviewing plugin changes when we get a chance. The changes here look good. I left one suggestion, but I think this is otherwise good to go. 

We don't have the infrastructure to test against but seeing as you are a codeowner I trust that you have vetted the changes work as you expect.

File: builder/ecs/run_config.go (hunk 1 above)
Comment (alexyueer) [reply builder/ecs/run_config.go]: thanks for your suggestion.

**Pull request merged**